examples/postgres/custom_arrays/README.md

```diff
@@ -0,0 +1,1162 @@
+* The array type follows the usual convention schema.type[].
+* The NOT NULL attribute means that the array itself must be set; Values inside the array still might be null.
+
+The Diesel team decided that array values are nullable by default because there so many things that may go wrong when
+dealing with array values thus making them nullable by default prevents a few potential crashes. Conversely, it also
+means you have to unwrap option types whenever you access data deserialized from an array.
+
+In total, the up.sql looks as below:
+
+```sql
+-- Your SQL goes here
+CREATE SCHEMA IF NOT EXISTS smdb;
+
+CREATE TYPE smdb.protocol_type AS ENUM (
+    'UnknownProtocol',
+    'GRPC',
+    'HTTP',
+    'UDP'
+);
+
+CREATE TYPE smdb.service_endpoint AS (
+	"name" Text,
+	"version" INTEGER,
+	"base_uri" Text,
+	"port" INTEGER,
+	"protocol" smdb.protocol_type
+);
+
+CREATE TABLE  smdb.service(
+	"service_id" INTEGER NOT NULL PRIMARY KEY,
+	"name" Text NOT NULL,
+	"version" INTEGER NOT NULL,
+	"online" BOOLEAN NOT NULL,
+	"description" Text NOT NULL,
+	"health_check_uri" Text NOT NULL,
+	"base_uri" Text NOT NULL,
+	"dependencies" INTEGER[] NOT NULL,
+	"endpoints" smdb.service_endpoint[] NOT NULL
+);
+```
+
+Now, it’s time to run the Diesel migration:
+
+```bash 
+diesel migration run
+```
+
+You may use a database console to double check if all types and tables have been created inside the custom schema. If a
+type somehow appears in the public default schema, double check if the type definition in up.sql has been prefixed with
+the schema name.
+
+The Diesel migration generates the following schema.rs file in the src root:
+
+```rust
+// @generated automatically by Diesel CLI.
+
+pub mod smdb {
+    pub mod sql_types {
+        #[derive(diesel::query_builder::QueryId, Clone, diesel::sql_types::SqlType)]
+        #[diesel(postgres_type(name = "service_endpoint", schema = "smdb"))]
+        pub struct ServiceEndpoint;
+    }
+
+    diesel::table! {
+        use diesel::sql_types::*;
+        use super::sql_types::ServiceEndpoint;
+
+        smdb.service (service_id) {
+            service_id -> Int4,
+            name -> Text,
+            version -> Int4,
+            online -> Bool,
+            description -> Text,
+            health_check_uri -> Text,
+            base_uri -> Text,
+            dependencies -> Array<Nullable<Int4>>,
+            endpoints -> Array<Nullable<ServiceEndpoint>>,
+        }
+    }
+}
+```
+
+Notice, the Postgres types are stored in Postgres therefore you are not seeing them in the generated schema. Only tables
+will show up in the generates schema. Furthermore, you will need a wrapper struct to map from Rust to the Postgres type.
+For the ServiceEndpoint, Diesel already generated a matching wrapper struct with the correct annotations. The service
+table then uses that wrapper struct “ServiceEndpoint” instead of the native Postgres type to reference the custom type
+in the endpoints array.
```

Comment:
```suggestion
For the ServiceEndpoint, Diesel already generated a matching zero sized SQL type struct with the correct annotations. The service
table then uses that SQL type `ServiceEndpoint` instead of the native Postgres type to reference the custom type
in the endpoints array.
```